embassy/src/lib.rs

```diff
@@ -1,5 +1,8 @@
 #![cfg_attr(not(any(feature = "std", feature = "wasm")), no_std)]
-#![cfg_attr(feature = "nightly", feature(generic_associated_types, type_alias_impl_trait))]
+#![cfg_attr(
+    feature = "nightly",
+    feature(generic_associated_types, type_alias_impl_trait, asm_experimental_arch)
```

Comment:
can you set `asm_experimental_arch` only when `feature = "nightly"` AND building for xtensa?